File: src/main/java/eu/execom/testutil/AssertPair.java
package eu.execom.testutil;

import eu.execom.testutil.enums.ObjectType;

public class AssertPair extends Pair {
    private ObjectType objectType;

    public AssertPair(final Object expected, final Object actual, final ObjectType objectType) {
        super(expected, actual);
        this.objectType = objectType;
    }

    public ObjectType getObjectType() {
        return objectType;
    }

    public void setObjectType(final ObjectType objectType) {
        this.objectType = objectType;
    }

}

File: src/main/scala/eu/execom/fabut/pair/AssertPair.scala
package eu.execom.fabut.pair

import eu.execom.fabut.AssertableType

/**
 * Represents expected & actual pair during asserting with
 * information of what type is the pair and if expected/actual
 * are fields of some already asserted object.
 */
case class AssertPair(path: String, expected: Any, actual: Any, assertableType: AssertableType.Value, property: Boolean = false) extends Pair


